### PR TITLE
Simplified credentials server

### PR DESCRIPTION
**Pull request by ericpromislow**
In order to take care of #2320 , first approve and merge this PR, 
then see #3096

Makes the credential server more like a tunnel, passing requests as-is to the selected `docker-credential-` helper.

### CODE REVIEW

File: src/go/docker-credential-none/cmd/root.go [hunk 1]
```diff
@@ -51,13 +52,22 @@ using the same format that docker uses when no credsStore field is specified in
...
+	DisableSuggestions: true,
+	SilenceUsage:       true,
 }
 
 // Execute adds all child commands to the root command and sets flags appropriately.
 // This is called by main.main(). It only needs to happen once to the rootCmd.
 func Execute() {
 	err := rootCmd.Execute()
 	if err != nil {
+		ptn := regexp.MustCompile(`unknown command "(.*?)" for "(?i:docker-credential-none(?:\.exe)?)"`)
```
Comment (mook-as) [inline src/go/docker-credential-none/cmd/root.go:64]: This seems like it doesn't belong in this PR.

File: src/main/credentialServer/README.md [hunk 2]
```diff
@@ -1,75 +0,0 @@
-# Credential Helper Server protocol
```
Comment (mook-as) [inline src/main/credentialServer/README.md:1]: We should still have a readme (though it doesn't need to describe the protocol anymore).

**Review by mook-as — CHANGES_REQUESTED** [review]

File: src/go/docker-credential-none/cmd/root.go (hunk 1 above)
Comment (ericpromislow) [reply src/go/docker-credential-none/cmd/root.go:64]: Actually, the PR is incomplete. It needs a test against an unknown endpoint, and that the error message is "standard". Which is why I made the above change.

Comment (ericpromislow) [reply src/go/docker-credential-none/cmd/root.go:64]: So hold off on reviewing for now....

File: src/main/credentialServer/README.md (hunk 2 above)
Comment (ericpromislow) [reply src/main/credentialServer/README.md]: Added a smaller one back, that basically references the upstream repository to read more about the protocol (which is defined there only in loose prose).

File: src/main/credentialServer/README.md [hunk 3]
```diff
@@ -2,74 +2,5 @@
...
-```
-/erase -d URL1
-
-RETURNS:
-
-HTTP Code 200
-
-Body: <EMPTY STRING>
-```
+See the docker-credentials-helper repository for more information on the protocol.
```
Comment (mook-as) [inline src/main/credentialServer/README.md:6]: This should link to the repository.

File: src/go/docker-credential-none/cmd/root.go [hunk 4]
```diff
@@ -51,13 +52,22 @@ using the same format that docker uses when no credsStore field is specified in
 This helper is intended for testing purposes, but will be used on Linux systems
 unless 'pass' and/or 'secretservice' is available.`,
 		configFileName),
+	DisableSuggestions: true,
+	SilenceUsage:       true,
```
Comment (mook-as) [inline src/go/docker-credential-none/cmd/root.go:56]: That seems wrong — usage should not be suppressed if there's a parse error (when being used interactively)?

I can see why that might be not great when using over the API, though…

File: src/go/docker-credential-none/cmd/root.go (hunk 1 above)
Comment (mook-as) [inline src/go/docker-credential-none/cmd/root.go:64]: Running it locally, it doesn't look like `.exe` ever shows up:
```
> .\docker-credential-none.exe aaa
Error: unknown command "aaa" for "docker-credential-none"
Run 'docker-credential-none --help' for usage.
Unknown credential action `aaa`
```

Also, note that we're still emitting the normal error _in addition to_ your custom one, rather than just the custom error; contrast with `wincred`:
```
> .\docker-credential-wincred.exe aaa
Unknown credential action `aaa`
```

(While I prefer not doing string matching, it doesn't look like that's reasonable given the cobra code:
https://github.com/spf13/cobra/blob/45e521ea93e2270ef7333487f0a87e3887de2da9/args.go#L22
We might need to just not use cobra here, instead…)

**Review by mook-as — CHANGES_REQUESTED** [review]
Hmm, this isn't going to be a security issue (because we pass the verb without going through shell), right?

It might be better to move the `docker-credential-none` changes to a separate PR instead.

File: src/main/credentialServer/README.md (hunk 3 above)
Comment (ericpromislow) [reply src/main/credentialServer/README.md:6]: I might as well -- I held up in case ownership changes, but that's not that likely.

File: src/go/docker-credential-none/cmd/root.go (hunk 1 above)
Comment (ericpromislow) [reply src/go/docker-credential-none/cmd/root.go:64]: I agree ... there is exactly one argument to process here, and cobra is getting in the way.

It might be a while before I get back to this, given the offline performance is a priority.

File: src/main/credentialServer/README.md (hunk 3 above)
Comment (ericpromislow) [reply src/main/credentialServer/README.md:6]: Done.

File: src/go/docker-credential-none/cmd/root.go (hunk 1 above)
Comment (ericpromislow) [reply src/go/docker-credential-none/cmd/root.go:64]: I added our own `os.Args` checker before handing it over to Cobra. The results look more conformant with the other helpers.

**Comment by ericpromislow** [comment]
Requested changes made

File: src/main/credentialServer/httpCredentialHelperServer.ts [hunk 5]
```diff
@@ -139,38 +120,33 @@ export class HttpCredentialHelperServer {
...
+      get:   requireJSONOutput,
+      erase: requireNoOutput,
+      store: requireNoOutput,
+    };
+    const checkerFn: checkerFnType|undefined = checkers[commandName];
+
+    if (request.method !== 'POST') {
+      requestCheckError = `Expecting a POST method for the credential-server list request, received ${ request.method }`;
+    } else if (!checkerFn) {
+      requestCheckError = `Unknown credential action '${ commandName }' for the credential-server, must be one of [${ Object.keys(checkers).join('|') }]`;
```
Comment (mook-as) [inline src/main/credentialServer/httpCredentialHelperServer.ts:142]: Should we always sort the command names? (Otherwise the order is technically undefined, and you're relying on the order in a test.)

**Review by mook-as — CHANGES_REQUESTED** [review]

**Review by mook-as — APPROVED** [review]
